src/test/util/logging.h

```diff
@@ -27,13 +24,16 @@ class DebugLogHelper
     //! (2) raising an error in check_found if no match was found
     //! Can return false to do the opposite in either case.
     using MatchFn = std::function<bool(const std::string* line)>;
-    MatchFn m_match;
-
-    void check_found();
 
-public:
     explicit DebugLogHelper(std::string message, MatchFn match = [](const std::string*){ return true; });
-    ~DebugLogHelper() noexcept(false) { check_found(); }
+
+    ~DebugLogHelper();
+
+private:
+    const std::string m_message;
+    bool m_found{false};
+    std::list<std::function<void(const std::string&)>>::iterator m_print_connection;
```

Comment:
nit/unrelated: I'm surprised to see an `std::list` here, `std::vector` is likely faster even for random erases